llvm/include/llvm/Frontend/OpenMP/OMP.td

```diff
@@ -1033,39 +1087,46 @@ def OMP_Teams : Directive<"teams"> {
     VersionedClause<OMPC_ThreadLimit>,
   ];
   let association = AS_Block;
+  let category = CA_Executable;
 }
 def OMP_ThreadPrivate : Directive<"threadprivate"> {
   let association = AS_None;
+  let category = CA_Declarative;
 }
 def OMP_Tile : Directive<"tile"> {
   let allowedOnceClauses = [
     VersionedClause<OMPC_Sizes, 51>,
   ];
   let association = AS_Loop;
+  let category = CA_Executable;
 }
 def OMP_Unknown : Directive<"unknown"> {
   let isDefault = true;
   let association = AS_None;
+  let category = CA_Utility;
```

Comment:
Please could you link me to the source for this? I am having trouble finding it.

Reply:
There are only two directives that are categorized as `utility` in the spec: `error` and `nothing`.  `unknown` is not in the spec (it's our local invention), but since it serves a somewhat similar purpose, I categorized it as `utility` as well.

Another category that it could fit under is `informational`.  Let me know what you think.

Reply:
If it is your local invention I guess you can define it how you like. Is there any documentation upstream I can read about it?